src/Security/Security.php

```diff
@@ -1138,6 +1242,46 @@ public static function database_is_ready()
     }
 
     /**
+     * Resets the database_is_ready cache
+     */
+    public static function clear_database_is_ready()
+    {
+        self::$database_is_ready = null;
+        self::$force_database_is_ready = null;
```

Comment:
We need to reduce coupling between non-test code invoking test-code; Most test classes aren't even available if phpunit isn't installed. It has to be here for now, or moved to DB layer.